addon_service/tasks/invocation.py

```diff
@@ -1,29 +1,27 @@
-from asgiref.sync import (
-    async_to_sync,
-    sync_to_async,
-)
+import celery
+from asgiref.sync import sync_to_async
 from django.db import transaction
 
 from addon_service.addon_imp.instantiation import get_storage_addon_instance
 from addon_service.common.dibs import dibs
 from addon_service.common.invocation_status import InvocationStatus
 from addon_service.models import AddonOperationInvocation
+from addon_toolkit.interfaces.storage import StorageAddonImp
 from addon_toolkit.json_arguments import json_for_typed_value
 
 
 __all__ = (
     "perform_invocation__async",
     "perform_invocation__blocking",
-    # TODO: @celery.task(def perform_invocation__celery)
+    "perform_invocation__celery",
 )
 
 
-@sync_to_async
-def perform_invocation__async(invocation: AddonOperationInvocation) -> None:
-    # implemented as a sync function (for django transactions)
-    # wrapped in sync_to_async (to guarantee a running event loop)
+def perform_invocation__blocking(invocation: AddonOperationInvocation) -> None:
+    # implemented as a sync function for django transactions
     with dibs(invocation):  # TODO: handle dibs errors
         try:
+            assert issubclass(invocation.imp_cls, StorageAddonImp)
```

Comment:
nit: Enforcing `StorageAddonImp` in particular feels like creating tech debt. Would enforcing `AddonImp` be worthwhile? Is this check truly required?

Reply:
oh wait, i added that to mollify mypy, which errors at the next line (because existing tech debt) -- will add comment explaining and leave it in (unless objection)

Reply:
(turns out i found an objection as writing a comment turned into a whole thing; went with a `# type: ignore` instead)